example/lib/demo_scrollables.dart

```diff
@@ -0,0 +1,291 @@
+  const _HorizontalList({Key? key}) : super(key: key);
+
+  @override
+  State<_HorizontalList> createState() => _HorizontalListState();
+}
+
+class _HorizontalListState extends State<_HorizontalList> {
+  final _boundsKey = GlobalKey();
+  late final ScrollController _scrollController;
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollController = ScrollController();
+  }
+
+  @override
+  void dispose() {
+    _scrollController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: [
+        Positioned.fill(
+          child: LayoutBuilder(builder: (context, constraints) {
+            return ListView.builder(
+              controller: _scrollController,
+              scrollDirection: Axis.horizontal,
+              itemBuilder: (context, index) {
+                if (index == 1) {
+                  return _LeaderAndFollowerListItem(
+                    width: constraints.maxWidth,
+                    recalculateGlobalOffset: _scrollController,
+                    boundsKey: _boundsKey,
+                  );
+                }
+
+                return _EmptyListItem(
+                  width: constraints.maxWidth,
+                );
+              },
+            );
+          }),
+        ),
+        Positioned.fill(
+          child: IgnorePointer(
+            child: Padding(
+              padding: const EdgeInsets.all(100),
+              child: DecoratedBox(
+                key: _boundsKey,
+                decoration: BoxDecoration(
+                  color: Colors.green.withOpacity(0.10),
+                ),
+              ),
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _EmptyListItem extends StatelessWidget {
+  const _EmptyListItem({
+    Key? key,
+    this.width,
+    this.height,
+  }) : super(key: key);
+
+  final double? width;
+  final double? height;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: width ?? double.infinity,
+      height: height ?? double.infinity,
+      child: DecoratedBox(
+        decoration: BoxDecoration(
+          color: Colors.black.withOpacity(0.05),
+          border: Border.all(color: Colors.black, width: 1),
+        ),
+      ),
+    );
+  }
+}
+
+class _LeaderAndFollowerListItem extends StatefulWidget {
+  const _LeaderAndFollowerListItem({
+    Key? key,
+    this.width,
+    this.height,
+    required this.recalculateGlobalOffset,
+    required this.boundsKey,
+  }) : super(key: key);
+
+  final double? width;
+  final double? height;
+  final Listenable recalculateGlobalOffset;
+  final GlobalKey boundsKey;
+
+  @override
+  State<_LeaderAndFollowerListItem> createState() => _LeaderAndFollowerListItemState();
+}
+
+class _LeaderAndFollowerListItemState extends State<_LeaderAndFollowerListItem> {
+  final _anchor = LeaderLink();
+
+  late final FollowerBoundary? _viewportBoundary;
```

Comment:
IMO, it's a bit weird to declare a member as `late final` and assign a value to it in two different places (`initState` and `didUpdateWidget`).

I would expect that a `late final` member is assigned only in `initState`.